app/templates/users.hbs

```diff
@@ -4,27 +4,57 @@
       <label class="inline-label">
         {{t 'general.school'}}:
       </label>
+
       <div id="school-selection" class="inline-data">
         {{currentUser.model.school.title}}
       </div>
     </div>
+
     <div id='textfilter' class="filter">
-      {{input type='text' value=textFilter placeholder=placeholderValue}}
+      <input type="text" class="user-search" placeholder={{t 'user.searchUsers'}}
+        oninput={{action "changeValue" value="target.value"}} />
     </div>
   </div>
+
   <section class='resultslist'>
     <div class="resultslist-header">
-      <div class="resultslist-title">
-        <h2>{{t 'general.users'}}</h2>
+      <h2 class="user-title">{{t 'general.users'}}</h2>
+
+      <div class="pagination-top">
+        {{#if query}}
+          {{users-pagination-links prevPages=prevResults morePages=moreResults
+            getPrevPage='getPrevPage' getNextPage='getNextPage' arrows=true}}
+        {{else}}
+          {{users-pagination-links prevPages=prevPages morePages=morePages
+            getPrevPage='getPrevPage' getNextPage='getNextPage' arrows=true}}
+        {{/if}}
       </div>
-      <!-- <div class='resultslist-actions'>
-        <button {{action 'addUser'}}>{{t 'general.addNew'}}</button>
-      </div> -->
     </div>
+
     <div class='resultslist-list'>
-      {{user-list
-        users=sortedUsers
-      }}
+      {{#if query}}
+        {{#if searchResults.isPending}}
+          {{pulse-loader}}
+        {{else}}
+          {{#if searchResults.content}}
+            {{user-list users=searchResults.content}}
+          {{else}}
+            <span class="no-results">{{general.noResultsFound}}</span>
+          {{/if}}
+        {{/if}}
+      {{else}}
+        {{user-list users=users}}
+      {{/if}}
+    </div>
+
+    <div class="pagination-bottom">
+      {{#if query}}
```

Comment:
Yup. You could use a nap haha. The `users-pagination-links` in the bottom are referencing different computed properties.

`prevResults vs. prevPages` & `moreResults vs. morePages` (due to both route and search being paginated)